src/core/thread/mle.cpp

```diff
@@ -1290,10 +1290,63 @@ void Mle::HandleAttachTimer(Timer &aTimer)
     aTimer.Get<Mle>().HandleAttachTimer();
 }
 
+Error Mle::DetermineParentRequestType(ParentRequestType &aType) const
+{
+    // This method determines the Parent Request type to use during an
+    // attach cycle based on `mAttachMode`, `mAttachCounter` and
+    // `mParentRequestCounter`. This method MUST be used while in
+    // `kAttachStateParentRequest` state.
+    //
+    // On success it returns `kErrorNone` and sets `aType`. It returns
+    // `kErrorNotFound` to indicate that device can now transition
+    // to`kAttachStateParentRequest`(has already sent required number
```

Comment:
This method already requires `kAttachStateParentRequest`. What does it mean to transition to `kAttachStateParentRequest`?

```suggestion
    // to `kAttachStateParentRequest` (has already sent required number
```